integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,254 @@
+from typing import Any, AsyncGenerator, List, Dict
+import httpx
+from loguru import logger
+from port_ocean.context import event
+from port_ocean.context.ocean import ocean
+from port_ocean.utils import http_async_client
+
+PAGE_SIZE = 50
+
+class ClickUpClient:
+    def __init__(self, clickup_url: str, clickup_api_key: str) -> None:
+        """
+        Initializes an instance of the ClickUpClient class.
+
+        Args:
+            clickup_url (str): The base URL for the ClickUp API.
+            clickup_api_key (str): The API key for authenticating requests to the ClickUp API.
+        """
+        self.clickup_url = clickup_url
+        self.api_url = f"{self.clickup_url}/api/v2"
+        self.api_key = clickup_api_key
+
+        self.api_auth_header = {"Authorization": self.api_key}
+        self.client = http_async_client
+        self.client.headers.update(self.api_auth_header)
+        self.client.timeout = httpx.Timeout(30)  # 30 seconds timeout for requests
+
+    @staticmethod
+    def parse_projects_from_folders(folders: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+        """
+        Extracts and flattens 'Projects' lists from a list of folders.
+
+        Args:
+            folders: A list of dictionaries representing folders with lists.
+
+        Returns:
+            A single list of project dictionaries extracted from folders named "Projects".
+        """
+        folder_with_projects = [folder["lists"] for folder in folders]
+        flattened_projects = [project for sublist in folder_with_projects for project in sublist]
+        return flattened_projects
+
+    @staticmethod
+    def parse_project(project: Dict[str, Any], team_id: str) -> Dict[str, Any]:
+        """
+        Adds team ID to a project dictionary.
+
+        Args:
+            project (dict): A dictionary representing a project.
+            team_id (str): The team ID to which the project belongs.
+
+        Returns:
+            dict: The updated project dictionary with the team ID.
+        """
+        project["__team_id"] = team_id
+        return project
```

Comment:
I don't think you need a function to append the team id to the project you can easily destructure the project dictionary {**project, "__team_id": id}